### PR TITLE
WIP: Remove RAnalPlugin.jmpmid use ANAL_ARCHINFO_ALIGN instead

### PR DESCRIPTION
**Pull request by condret**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

**Comment by trufae** [comment]
Squash when merging pls

File: libr/anal/bb.c [hunk 1]
```diff
@@ -31,10 +31,13 @@ static bool bb_from_offset_first_cb(RAnalBlock *block, void *user) {
 }
 
 R_API RAnalBlock *r_anal_bb_from_offset(RAnal *anal, ut64 off) {
-	if (anal->opt.jmpmid && anal->cur->jmpmid) {
-		BBFromOffsetJmpmidCtx ctx = { off, NULL };
-		r_anal_blocks_foreach_in (anal, off, bb_from_offset_jmpmid_cb, &ctx);
-		return ctx.ret;
+	if (anal->opt.jmpmid) {
+		const int align = r_anal_archinfo (anal, R_ANAL_ARCHINFO_ALIGN);
+		if (align <= 1 || !(off % align)) {
```
Comment (trufae) [inline libr/anal/bb.c:36]: what about adding a function like: `if (r_anal_is_aligned (anal, off)) {` and reuse it in every call.

Comment (condret) [reply libr/anal/bb.c:36]: :thinking:

File: libr/anal/p/anal_gb.c [hunk 2]
```diff
@@ -1562,7 +1562,7 @@ static int archinfo(RAnal *anal, int q) {
 	case R_ANAL_ARCHINFO_MIN_OP_SIZE:
 		return 1;
 	}
-	return 1;
+	return -1;
```
Comment (trufae) [inline libr/anal/p/anal_gb.c:1565]: uhm, -1 can result in funky behaviours, isnt? youa re now checking for <=1 so it sohuld be good. but maybe better to give 1 or 0 imho

Comment (condret) [reply libr/anal/p/anal_gb.c:1565]: but 1 is valid alignment. -1 indicates unsupported query

**Pull request merged**